src/common/Utilities/StringConvert.h

```diff
@@ -0,0 +1,178 @@
+        {
+            std::string buf(20,0); /* largest unsigned 64-bit value is 20 decimal characters, largest signed 64-bit value is 19 */
+            char* const start = buf.data();
+            char* const end = (start + buf.length());
+            std::to_chars_result const res = std::to_chars(start, end, val);
+            ASSERT(res.ec == std::errc());
+            buf.resize(res.ptr - start);
+            return buf;
+        }
+    };
+
+#ifdef TRINITY_NEED_CHARCONV_WORKAROUND
+    /*
+        If this is defined, std::from_chars will cause linkage errors for 64-bit types.
+        (This is a bug in clang-7.)
+
+        If the clang requirement is bumped to >= clang-8, remove this ifdef block and its
+        associated check in cmake/compiler/clang/settings.cmake
+    */
+    template <>
+    struct For<uint64, void>
+    {
+        static Optional<uint64> FromString(std::string_view str, int base = 10)
+        {
+            if (str.empty())
+                return std::nullopt;
+            try
+            {
+                size_t n;
+                uint64 val = std::stoull(std::string(str), &n, base);
+                if (n != str.length())
+                    return std::nullopt;
+                return val;
+            }
+            catch (...) { return std::nullopt; }
+        }
+
+        static std::string ToString(uint64 val)
+        {
+            return std::to_string(val);
+        }
+    };
+
+    template <>
+    struct For<int64, void>
+    {
+        static Optional<int64> FromString(std::string_view str, int base = 10)
+        {
+            try {
+                if (str.empty())
+                    return std::nullopt;
+                size_t n;
+                int64 val = std::stoll(std::string(str), &n, base);
+                if (n != str.length())
+                    return std::nullopt;
+                return val;
+            }
+            catch (...) { return std::nullopt; }
+        }
+
+        static std::string ToString(int64 val)
+        {
+            return std::to_string(val);
+        }
+    };
+#endif
+
+    template <>
+    struct For<bool, void>
+    {
+        static Optional<bool> FromString(std::string_view str)
+        {
+            if ((str == "1") || StringEqualI(str, "y") || StringEqualI(str, "on") || StringEqualI(str, "yes") || StringEqualI(str, "true"))
+                return true;
+            if ((str == "0") || StringEqualI(str, "n") || StringEqualI(str, "off") || StringEqualI(str, "no") || StringEqualI(str, "false"))
+                return false;
+            return std::nullopt;
+        }
+
+        static std::string ToString(bool val)
+        {
+            return (val ? "1" : "0");
+        }
+    };
+}
+
+namespace Trinity
+{
+    template <typename Result, typename... Params>
+    Optional<Result> StringTo(std::string_view str, Params&&... params) { return Trinity::Impl::StringConvertImpl::For<Result>::FromString(str, std::forward<Params>(params)...); }
```

Comment:
line returns please, same L175

Reply:
in my defense that line was a lot simpler when i initially added it...

it kind of just kept growing

Reply:
Yeah templates do that